include/openthread/link.h

```diff
@@ -496,6 +496,14 @@ bool otLinkIsEnergyScanInProgress(otInstance *aInstance);
  */
 otError otLinkSendDataRequest(otInstance *aInstance);
 
+/**
+ * This function enqueues an MLE ChildUpdateRequest message for transmission.
+ *
+ * @param[in] aInstance  A pointer to an OpenThread instance.
+ *
+ */
+void otLinkSendChildUpdateRequest(otInstance *aInstance);
```

Comment:
Is the intention to provide an API for the application so that it can send keep-alive and don't get CSL timeout? If so we can do that directly by configuring the poll period. For example, we set the poll period by `otLinkSetPollPeriod` and it will send Data Request regularly. (Looks like it degrade to an SED)

If we just want to send one data request, we can use `otLinkSendDataRequest`. I think the best way is to send something just for keep CSL alive is to use `otLinkSendEmptyData` because that sends an empty 802.15.4 frame containing CSL IE which has the least overhead. These two APIs are available if we enable the option `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE`.

Reply:
The intention was not to use ChildUpdateRequest for keep-alive, but allow the application to validate the link periodically "to avoid cases where the MAC Data Request is acknowledged erroneously by the receiver" according to Thread Specification.

Reply:
Could you point me to the related information in the spec? I'd like to know more about the context. My version is probably old and I didn't find related content.

Reply:
You find below sentence in Chapter **4.6.3 Timing Out Children** (Specification v1.3)
![image](https://user-images.githubusercontent.com/30142883/221516222-a29f9549-045b-493e-a548-d9b59e168984.png)
Also v1.1 had a similar sentence in Chapter 4.7.5, back then leaving the time intervals specifically up to the application.

Reply:
Thanks! My spec is indeed old and I need to update it.

Regarding this issue, the 'SHOULD' here recommends the stack to do the practice. Currently OT doesn't send Child Update Request periodically. I'm ok to adopt the practice. Then we should directly call `ScheduleChildUpdateRequest` somewhere in the code, instead of providing an API to the application.

Reply:
I think we need more discussion if we want to add periodical Child Update Request in OT. For example, how long should we send it and how we implement it? So I suggest we do this in a separate PR.

Reply:
Ok thank you, I have removed those lines from this PR.